components/brave_new_tab_ui/api/initialData.ts

```diff
@@ -37,3 +49,61 @@ export default async function getInitialData (): Promise<InitialData> {
     throw Error('Error getting initial data')
   }
 }
+
+export async function getRewardsPreInitialData (): Promise<PreInitialRewardsData> {
+  try {
+    const [
+      enabledAds,
+      enabledMain
+    ] = await Promise.all([
+      new Promise(resolve => chrome.braveRewards.getAdsEnabled((enabledAds: boolean) => {
+        resolve(enabledAds)
+      })),
+      new Promise(resolve => chrome.braveRewards.getRewardsMainEnabled((enabledMain: boolean) => {
+        resolve(enabledMain)
+      }))
+    ])
+    return {
+      enabledAds,
+      enabledMain
+    } as PreInitialRewardsData
+  } catch (err) {
+    throw Error('Could not retrieve initial rewards data: ' + err)
+  }
+}
+
+export async function getRewardsInitialData (): Promise<InitialRewardsData> {
+  try {
+    const [
+      onlyAnonWallet,
+      adsEstimatedEarnings,
+      reports,
+      balance
+    ] = await Promise.all([
+      new Promise(resolve => chrome.braveRewards.onlyAnonWallet((onlyAnonWallet: boolean) => {
+        resolve(!!onlyAnonWallet)
+      })),
+      new Promise(resolve => chrome.braveRewards.getAdsEstimatedEarnings((adsEstimatedEarnings: number) => {
+        resolve(adsEstimatedEarnings)
+      })),
+      new Promise(resolve => chrome.braveRewards.getBalanceReports((reports: Record<string, NewTab.RewardsReport>) => {
+        resolve(reports)
+      })),
+      new Promise(resolve => chrome.braveRewards.fetchBalance((balance: NewTab.RewardsBalance) => {
+        resolve(balance)
+      })),
+      new Promise(resolve => {
+        chrome.braveRewards.getGrants()
+        resolve(true)
+      })
+    ])
+    return {
+      onlyAnonWallet,
+      adsEstimatedEarnings,
+      reports,
+      balance
+    } as InitialRewardsData
+  } catch (err) {
+    throw Error('Could not retrieve initial rewards data: ' + err)
```

Comment:
`, err` will ensure that the entire error gets logged, even if it's not a string and including the stack.
Although I'm not sure what errors could happen here since we're never calling `reject` on any of the promises. Do we know if these APIs return async errors in their callbacks, or do they actually cause runtime errors when called?

Reply:
good catch, thx. from the rewards api the

error are returned like: `return RespondNow(Error(errString))` in the callback functions for `chrome.braveRewards`